### PR TITLE
Fix -Werror=type-limits.

### PR DESCRIPTION
**Pull request by bdice**
## Description
I'm compiling cuDF as a part of another application and ran into errors from `-Werror=type-limits`. There are a few comparisons between unsigned types like `value < 0`, which is never true. This PR removes those impossible code paths.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by bdice** [comment]
I'm not sure how to enforce `-Werror=type-limits` and would appreciate guidance if that is something we can do in all libcudf builds.

File: cpp/src/io/comp/cpu_unbz2.cpp [hunk 1]
```diff
@@ -221,7 +221,7 @@ int32_t bz2_decompress_block(unbz_state_s* s)
   if (getbits(s, 1)) return BZ_DATA_ERROR;  // blockRandomized not supported (old bzip versions)
 
   s->origPtr = getbits(s, 24);
-  if (s->origPtr < 0 || s->origPtr > 10 + 100000 * s->blockSize100k) return BZ_DATA_ERROR;
+  if (s->origPtr > 10 + 100000 * s->blockSize100k) return BZ_DATA_ERROR;
```
Comment (mythrocks) [inline cpp/src/io/comp/cpu_unbz2.cpp:224]: 👍 One wonders how we got here in the first place.

**Review by mythrocks — APPROVED** [review]
LGTM!

**Review by davidwendt — APPROVED** [review]

**Comment by vyasr** [comment]
> I'm not sure how to enforce `-Werror=type-limits` and would appreciate guidance if that is something we can do in all libcudf builds.

Maybe I'm misunderstanding the problem, but could we just add this flag to our builds? We can add flags like this [here](https://github.com/rapidsai/cudf/blob/branch-24.06/cpp/cmake/Modules/ConfigureCUDA.cmake#L16).

**Pull request merged**